staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -117,28 +117,35 @@ type listenAddress struct {
 	failureMode string
 }
 
+// parseAddresses parses, validates addresses and removes duplicates
 func parseAddresses(addressesToParse []string) ([]listenAddress, error) {
-	var addresses []listenAddress
+	var keyOrder []string
 	parsed := make(map[string]listenAddress)
 	for _, address := range addressesToParse {
 		if address == "localhost" {
-			ip := listenAddress{address: "127.0.0.1", protocol: "tcp4", failureMode: "all"}
-			parsed[ip.address] = ip
-			ip = listenAddress{address: "::1", protocol: "tcp6", failureMode: "all"}
-			parsed[ip.address] = ip
+			ipv4 := listenAddress{address: "127.0.0.1", protocol: "tcp4", failureMode: "all"}
+			ipv6 := listenAddress{address: "::1", protocol: "tcp6", failureMode: "all"}
+			parsed[ipv4.address] = ipv4
+			parsed[ipv6.address] = ipv6
+			// IPv4 address for localhost must go first for backward compatibility
+			keyOrder = append(keyOrder, ipv4.address, ipv6.address)
```

Comment:
nit: you could add two lines of
```
keyOrder = append(keyOrder, ip.address)
```
instead of changing existing lines. I'm not sure which way is more readable.